### PR TITLE
 [MRG] Fix: Added Axes object check in `plot_cells`

### LINKED ISSUE #735: `plot_cells` error when users pass in a generic matplotlib.Axes instance
Found by @wagdy88.

If you create a figure with multiple subplots and then pass one of those subplots into `plot_cells()` (as can be done for most plotting functions in `viz.py`), an error is thrown.

For example, running
```python
from hnn_core import jones_2009_model
net = jones_2009_model()
import matplotlib.pyplot as plt
fig, axes = plt.subplots(1, 1)
net.plot_cells(ax=axes)
```
produces the following:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 5
      3 import matplotlib.pyplot as plt
      4 fig, axes = plt.subplots(1, 1)
----> 5 net.plot_cells(ax=axes)

File ~/hnn-core/hnn_core/network.py:1418, in Network.plot_cells(self, ax, show)
   1402 def plot_cells(self, ax=None, show=True):
   1403     """Plot the cells using Network.pos_dict.
   1404 
   1405     Parameters
   (...)
   1416         The matplotlib figure handle.
   1417     """
-> 1418     return plot_cells(net=self, ax=ax, show=show)

File ~/hnn-core/hnn_core/viz.py:598, in plot_cells(net, ax, show)
    596         color = colors[cell_type]
    597         marker = markers[cell_type]
--> 598         ax.scatter(x, y, z, c=color, s=50, marker=marker, label=cell_type)
    600 if net.rec_arrays:
    601     cols = plt.get_cmap('inferno', len(net.rec_arrays) + 2)

File ~/anaconda3/envs/hnn_core/lib/python3.9/site-packages/matplotlib/__init__.py:1465, in _preprocess_data.<locals>.inner(ax, data, *args, **kwargs)
   1462 @functools.wraps(func)
   1463 def inner(ax, *args, data=None, **kwargs):
   1464     if data is None:
-> 1465         return func(ax, *map(sanitize_sequence, args), **kwargs)
   1467     bound = new_sig.bind(ax, *args, **kwargs)
   1468     auto_label = (bound.arguments.get(label_namer)
   1469                   or bound.kwargs.get(label_namer))

TypeError: scatter() got multiple values for argument 's'
```

I'm pretty sure the issue here is that Matplotlib is trying to project a 3D plot onto a 2D Axes instance, however, this can be confusing to users who aren't familiar with using Matplotlib to plot in 3D. I suggest that we either come up with a patch within `plot_cells` that converts the Axes projection from 2D to 3D (thus allowing the user to pass in any generic Axes object and have it work) or remove the `ax` parameter from `plot_cells` altogether to remove confusion.

### PR DESCRIPTION
**Pull request by samadpls**
Closes #735 
- Added check for invalid Axes object in `plot_cells` function, raises an error if found.

### CODE REVIEW

File: hnn_core/viz.py [hunk 1]
```diff
@@ -577,12 +577,15 @@ def plot_cells(net, ax=None, show=True):
...
     import matplotlib.pyplot as plt
-    from mpl_toolkits.mplot3d import Axes3D  # noqa: F401 unused import
+    from mpl_toolkits.mplot3d import Axes3D
 
     if ax is None:
         fig = plt.figure()
         ax = fig.add_subplot(111, projection='3d')
 
+    elif not isinstance(ax, Axes3D):
+        raise TypeError("The 'ax' argument must be an instance of Axes3D.")
```
Comment (jasmainak) [inline hnn_core/viz.py:587]: when raising error messages, it is good practice to say what you got:

"Expected blah, got blahblah"

You save the user some guesswork by doing this ... also helps users without advanced debugging capabilities

**Comment by jasmainak** [comment]
diff looks good. Please set PR to [MRG] once it's ready on your end and update what's new

perhaps it's time to address a slightly more complicated issue @samadpls ? perhaps: https://github.com/jonescompneurolab/hnn-core/issues/658 ?

File: doc/whats_new.rst [hunk 2]
```diff
@@ -40,6 +40,9 @@ Changelog
   returned for cases when there should be no valid matches, by `George Dang`_
   in :gh:`739`
 
+- Added check for invalid Axes object in :func:`~hnn_core/viz/plot_cells` 
+  function, by `Abdul Samad Siddiqui`_ in :gh:`744`
```
Comment (rythorpe) [inline doc/whats_new.rst:44]: You'll also want to link your name to your GH account at the [bottom of the file](https://github.com/jonescompneurolab/hnn-core/blob/54bd27805198f8c345efe307b2860ad271f1dc3f/doc/whats_new.rst?plain=1#L469).

File: doc/whats_new.rst [hunk 3]
```diff
@@ -43,6 +43,9 @@ Changelog
...
```
Comment (rythorpe) [inline doc/whats_new.rst:47]: Nitpick, but we should probably be consistent with punctuation (which none of us have been good about so far).

```suggestion
  function, by `Abdul Samad Siddiqui`_ in :gh:`744`.
```

**Review by rythorpe — APPROVED** [review]

**Comment by rythorpe** [comment]
Thanks @samadpls!

**Pull request merged**